cms/djangoapps/models/settings/course_metadata.py

```diff
@@ -125,8 +130,9 @@ def fetch(cls, descriptor):
         """
         result = {}
         metadata = cls.fetch_all(descriptor)
+        filtered_list = cls.filtered_list()
         for key, value in metadata.iteritems():
-            if key in cls.filtered_list():
+            if key in filtered_list:
```

Comment:
@itsjeyd I was thinking about this change. Basically everytime that "if" is called it calls that method 20+ times.  Reading through the other uses of this method, its used in the same way, and looking at the code there is no reason why this shouldn't be a valid change.

What's the issue you're thinking of?

Reply:
@macornwell It's definitely a valid change! What I meant when I said "Is there a specific reason for making this change?" was whether the functionality that this PR introduces depends on this change somehow. This doesn't seem to be the case, so I would prefer to revert this change for the following reason:

It will be a while until we merge the `adaptive-learning-master` branch into upstream `master`. To avoid running into compatibility issues later on, we will have to keep rebasing the `adaptive-learning-master` branch on upstream `master` on a regular basis. The more changes a branch contains, the higher the potential for rebasing conflicts -- especially if a PR branch makes lots of changes to existing files/core code. So it's good practice to avoid *drive-by changes* such as the one you made here: Yes, the change does reduce the number of times that `filtered_list` gets called, which is a good thing. But the previous version of the code didn't cause any scalability issues up until now (if that were the case it would have been changed a long time ago). So it's not really worth increasing the potential for rebasing conflicts, just to be able to keep this change.

Does that make sense?

Reply:
I completely get what you mean now. That hadn't crossed my mind at all! Thanks for taking the time to explain it.

Reply:
@macornwell No worries! Most newcomers run into this at some point, I know I certainly did :)